plugin.py

```diff
@@ -299,6 +302,8 @@ def refresh_layers(self):
                             required_layer.layer = layer
                             self.iface.legendInterface().setLayerVisible(
                                 layer, True)
+                            if self.crs:
+                                layer.setCrs(self.crs)
```

Comment:
Ok.  That's fine. We assume that the user selects a CRS that corresponds to his database in the backend.  Later we might need to add rules to check what database CRS has already been assigned to a table and compare this with the selected CRS so that a user can not select a CRS that is not the same as his database